### PR TITLE
Ytb 52 contract tests for api

### PR DESCRIPTION
**Pull request by sennikrauss**

### CODE REVIEW

**Review by rowolff — COMMENTED** [review]
Good job! This was a tough one to crack! I added some comments and suggestions for you to take into consideration but it's not a must-do.

Some more questions:
1. When I run it locally it fails because redis is not there (I don't have the docker image running) - you should also try to make the tests pass without any dependency to redis. Can you do that?
2. I did not find the part in the CI where the tests are run - can you please point me to that?

File: packages/idealo.design/src/ui/pages/BlogPage/consumer-contract.spec.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ const PORT = 4000;
 const URL = 'http://localhost:';
 
 const provider = new Pact({
-    consumer: 'Consumer for blogPage',
-    provider: 'Provider for functions in ./data (BlogPage)',
+    consumer: 'BlogTestingConsumer',
+    provider: 'BlogTestingProvider',
```
Comment (rowolff) [inline packages/idealo.design/src/ui/pages/BlogPage/consumer-contract.spec.js:21]: 👍

File: packages/idealo.design/src/server/server.js [hunk 2]
```diff
@@ -111,15 +115,12 @@ if (CLIENT_ID) {
...
 
-process.env.dangerousTestModeArgument = 'false';
-
 function isAuthenticated(req, res, next) {
-  if(req.session.user){
-    return next();
-  }
-  if(process.env.dangerousTestModeArgument==='true'){
+  const testUser = req.header('authorization')
+  if(req.session.user || testUser){
```
Comment (rowolff) [inline packages/idealo.design/src/server/server.js:120]: much better 👍  

maybe you can even supply the `req.session.user` in the request filter in the provider test? If you manage to do that, you can leave the isAuthenticated function as simple as it was before :)

**Review by MuellerBettina — COMMENTED** [review]
Hello @nforgerit 😊 Could you please check [our additions to the deploy.yml](https://github.com/idealo/nwp/pull/86/commits/e24d53d0477a3fb3697e1d23b8a88ef6106aa1d0) file specifically? We're not sure if our current implementation would stop the deployment in case the tests fail. Thank you!

**Comment by rowolff** [comment]
@MuellerBettina yes the pipeline should abort if the tests fail. Please remove the multiple node versions in the stategy.matrix! You don't want to run the tests 3 times for 3 different node versions. Pick a node version that matches your deployment target and then set up the runner to run with this specific version

**Review by rowolff — APPROVED** [review]
Well done - this was a hard one :)

**Pull request merged**